client/data/marketplace/use-wporg-plugin-query.ts

```diff
@@ -50,3 +61,41 @@ export const useWPORGPlugins = (
 		}
 	);
 };
+
+const extractPages = ( pages: Array< { plugins: object; info: object } > = [] ) =>
+	pages.flatMap( ( page ) => page.plugins ).map( normalizePluginData );
+
+export const useWPORGInfinitePlugins = (
+	options: WPORGOptionsType,
+	{ enabled = true, staleTime = 1000 * 60 * 60 * 2, refetchOnMount = true }: UseQueryOptions = {}
+): UseQueryResult => {
+	const [ search, author ] = extractSearchInformation( options.searchTerm );
+	const locale = useSelector( getCurrentUserLocale );
+
+	return useInfiniteQuery(
+		getPluginsListKey( options, true ),
+		( { pageParam = 1 } ) =>
+			fetchPluginsList( {
+				pageSize: options.pageSize,
+				page: pageParam,
+				category: options.category,
+				locale: options.locale || locale,
+				search,
+				author,
+			} ),
+		{
+			select: ( data: InfiniteData< { plugins: object; info: { page: number } } > ) => {
+				return {
+					...data,
+					plugins: extractPages( data.pages ),
+				};
+			},
+			getNextPageParam: ( lastPage ) => {
+				return ( lastPage.info.page || 0 ) + 1;
+			},
+			enabled: enabled,
+			staleTime: staleTime,
+			refetchOnMount: refetchOnMount,
+		}
+	);
+};
```

Comment:
This config differs only slightly from the one in `useWPORGPlugins`:
https://github.com/Automattic/wp-calypso/blob/6e290cc85af1a81e7196b4c2812eb411aa06dd75/client/data/marketplace/use-wporg-plugin-query.ts#L35-L63

Based on the [react query docs](https://react-query.tanstack.com/reference/useInfiniteQuery), would it be possible to define the query config only once with the infinite scroll, and optionally use it without the `fetchNextPage` callback?